lib/sprockets/less/v3/importer.rb

```diff
@@ -0,0 +1,224 @@
+          relatives = paths.map { |path_detected| path_detected.to_s.start_with?('.') ? Pathname.new(path_detected) : Pathname.new(path_detected.to_s.prepend('./')) }
+          context.environment.paths.each do |load_path|
+            relative_path = Pathname.new(base_path).relative_path_from(Pathname.new(load_path)).join(path)
+            paths.unshift(relative_path, partialize_path(relative_path))
+          end
+          paths = paths.unshift(relatives)
+          paths
+        end
+
+        def filtered_processor_classes
+          classes = super
+          classes << Sprockets::Preprocessors::DefaultSourceMap if defined?(Sprockets::Preprocessors::DefaultSourceMap)
+          classes << Sprockets::SourceMapProcessor if defined?(Sprockets::SourceMapProcessor)
+          classes
+        end
+
+        def content_type_of_path(context, path)
+          attributes = context.environment.send(:parse_path_extnames, path.to_s)
+          content_type = attributes[1]
+          [content_type, attributes]
+        end
+
+        def build_input_for_process(context, path, data)
+          content_type, _attributes = content_type_of_path(context, path)
+          {
+            environment: context.environment,
+            cache: context.environment.cache,
+            uri: path.to_s,
+            filename: path.to_s,
+            load_path: context.environment.paths,
+            name: File.basename(path),
+            content_type: content_type,
+            data: data,
+            metadata: context.metadata
+          }
+        end
+
+        def check_path_before_process(_context, path)
+          path
+        end
+
+        def call_processor_input(processor, context, input, processors)
+          metadata = (input[:metadata] || {}).dup
+          metadata[:data] = input[:data]
+          result = processor.call(input)
+          processors.delete(processor)
+          handle_process_result(context, result, processors, metadata)
+        end
+
+        def handle_complex_process_result(context, result, processors)
+          data = result[:data] if result.key?(:data)
+          context.metadata.merge!(result)
+          context.metadata.delete(:data)
+          if result.key?(:required)
+            result[:required].each do |file|
+              file_asset = context.environment.load(file)
+              data += process(processors, context, file_asset.filename)
+            end
+          end
+          data
+        end
+
+        def handle_process_result(context, result, processors, metadata)
+          data = nil
+          case result
+            when NilClass
+              data = metadata[:data]
+            when Hash
+              data = handle_complex_process_result(context, result, processors)
+            when String
+              data = result
+            else
+              raise Error, "invalid processor return type: #{result.class}"
+          end
+          data
```

Comment:
Hmm. We should just skip the `data` var assignment altogether and use `case ... end` as a return value.